client/src/features/project/components/ProjectSettingsDescription.tsx

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2023 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { Card, CardBody, FormText, Input, Label } from "reactstrap";
+
+import { Loader } from "../../../components/Loader";
+import { InlineSubmitButton } from "../../../components/buttons/Button";
+import { useGetProjectIndexingStatusQuery } from "../projectKgApi";
+import { useProjectMetadataQuery } from "../../projects/projectsKgApi";
+import { useUpdateDescriptionMutation } from "../projectCoreApi";
+import { CoreErrorAlert } from "../../../components/errors/CoreErrorAlert";
+import { CoreErrorContent } from "../../../utils/definitions";
+import { SettingRequiresKg } from "./ProjectSettingsUtils";
+
+interface ProjectSettingsDescriptionProps {
+  gitUrl: string;
+  projectFullPath: string;
+  projectId: number;
+  settingsReadOnly?: boolean;
+}
+export function ProjectSettingsDescription({
+  gitUrl,
+  projectFullPath,
+  projectId,
+  settingsReadOnly = false,
+}: ProjectSettingsDescriptionProps) {
+  const [description, setDescription] = useState("");
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
+    skip: !projectFullPath || !projectId,
+  });
+  const projectMetadata = useProjectMetadataQuery(
+    { projectPath: projectFullPath, projectId },
+    {
+      skip:
+        !projectFullPath ||
+        !projectId ||
+        !projectIndexingStatus.data?.activated,
+    }
+  );
+  const [updateDescriptionMutation, updateDescriptionStatus] =
+    useUpdateDescriptionMutation();
+  const updateDescription = useCallback(
+    (description: string) =>
+      updateDescriptionMutation({
+        description,
+        gitUrl,
+        projectId,
+      }),
+    [updateDescriptionMutation, gitUrl, projectId]
+  );
+
+  const submit = () => {
+    updateDescription(description);
+  };
```

Comment:
Create only one callback function here:

```suggestion
  const onSubmit = useCallback(() => {
    updateDescriptionMutation({
      description,
      gitUrl,
      projectId,
    });
  }, [description, gitUrl, projectId, updateDescriptionMutation]);
```

Reply:
Could you briefly explain why this is a better solution? Just to be more concise or is there any impact on performances?

Reply:
It's simpler, and it's less code to read. Performance-wise, it's also better since doing `onClick={onSubmit}` will be a stable reference.